[DesktopGL] Playing a Song fails upon second play on OSX

Hello,

I'm trying to port our game to DesktopGL and so far it works well on Windows & Linux but crashes on OS X when ```MediaPlayer.Play()``` is called a second time (with a different ```Song```) with the following stack trace (and the exact same build / files):

```
Unhandled Exception:
System.InvalidOperationException: Invalid Name
  at MonoGame.Utilities.ALHelper.Check () <0xa8db610 + 0x0003f> in <filename unknown>:0 
  at MonoGame.Utilities.OggStream.set_Volume (Single value) <0xa8db4f8 + 0x00043> in <filename unknown>:0 
  at Microsoft.Xna.Framework.Media.Song.set_Volume (Single value) <0xa9e35d8 + 0x00023> in <filename unknown>:0 
  at Microsoft.Xna.Framework.Media.MediaPlayer.PlatformPlaySong (Microsoft.Xna.Framework.Media.Song song, Nullable`1 startPosition) <0xa9e3490 + 0x000a3> in <filename unknown>:0 
  at Microsoft.Xna.Framework.Media.MediaPlayer.PlaySong (Microsoft.Xna.Framework.Media.Song song, Nullable`1 startPosition) <0xa9e3430 + 0x00027> in <filename unknown>:0 
  at Microsoft.Xna.Framework.Media.MediaPlayer.Play (Microsoft.Xna.Framework.Media.Song song, Nullable`1 startPosition) <0xa9e3250 + 0x000a7> in <filename unknown>:0 
  at Microsoft.Xna.Framework.Media.MediaPlayer.Play (Microsoft.Xna.Framework.Media.Song song) <0xa9e3200 + 0x0003b> in <filename unknown>:0 
```

I've narrowed this problem a bit more by adding ```ALHelper.Check()``` after every AL command, and the error seems to appear here:

```
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: Invalid Name
  at MonoGame.Utilities.ALHelper.Check () <0x5ecbb70 + 0x00047> in <filename unknown>:0 
  at MonoGame.Utilities.OggStream.Stop () <0xbf38620 + 0x0002f> in <filename unknown>:0 
  at Microsoft.Xna.Framework.Media.Song.Stop () <0xbf385f0 + 0x00017> in <filename unknown>:0 
  at Microsoft.Xna.Framework.Media.MediaQueue.Clear () <0x9abec30 + 0x0003b> in <filename unknown>:0 
  at Microsoft.Xna.Framework.Media.MediaPlayer.Play (Microsoft.Xna.Framework.Media.Song song, Nullable`1 startPosition) <0x9abeae0 + 0x00053> in <filename unknown>:0 
  at Microsoft.Xna.Framework.Media.MediaPlayer.Play (Microsoft.Xna.Framework.Media.Song song) <0x9abea90 + 0x0003b> in <filename unknown>:0 
```

Note that I load/unload ```Song``` on demand. Before loading & playing the next ```Song```, I call ```MediaPlayer.Stop()``` and ```Content.Unload()```. The error is occurring while clearing the previous queue (which has been stopped and disposed) on ```var state = AL.GetSourceState(alSourceId);``` in ```OggStream.Stop()```.

It looks like OS X OpenAL implementation handles ```AL.GetSourceState``` differently. It seems to raise an error flag if ```alSourceId``` is disposed, while on other systems it returns ```ALSourceState.Stopped``` with no error. Hence, calling ```OggStream.Stop()``` should be prevented if the stream has already been disposed.

I've tested that and it works just great. Going to submit PR.

--- a/MonoGame.Framework/Media/Song.NVorbis.cs
+++ b/MonoGame.Framework/Media/Song.NVorbis.cs
@@ -31,11 +31,18 @@ namespace Microsoft.Xna.Framework.Media
 		
         void PlatformDispose(bool disposing)
         {
+            if (stream == null)
+                return;
+
             stream.Dispose();
+            stream = null;
         }
 
         internal void Play(TimeSpan? startPosition)
         {
+            if (stream == null)
+                return;
+
             stream.Play();
             if (startPosition != null)
                 stream.SeekToPosition((TimeSpan)startPosition);
@@ -45,27 +52,42 @@ namespace Microsoft.Xna.Framework.Media
 
         internal void Resume()
         {
+            if (stream == null)
+                return;
+
             stream.Resume();
         }
 
         internal void Pause()
         {
+            if (stream == null)
+                return;
+
             stream.Pause();
         }
 
         internal void Stop()
         {
+            if (stream == null)
+                return;
+
             stream.Stop();
             _playCount = 0;
         }
 
         internal float Volume
         {
-            get { return _volume; }
+            get
+            {
+                if (stream == null)
+                    return 0.0f;
+                return _volume; 
+            }
             set
             {
                 _volume = value;
-                stream.Volume = _volume;
+                if (stream == null)
+                    stream.Volume = _volume;
             }
         }
 
@@ -73,6 +95,8 @@ namespace Microsoft.Xna.Framework.Media
         {
             get
             {
+                if (stream == null)
+                    return TimeSpan.FromSeconds(0.0);
                 return stream.GetPosition();
             }
         }

--- a/MonoGame.Framework/Media/Song.NVorbis.cs
+++ b/MonoGame.Framework/Media/Song.NVorbis.cs
@@ -86,7 +86,7 @@ namespace Microsoft.Xna.Framework.Media
             set
             {
                 _volume = value;
-                if (stream == null)
+                if (stream != null)
                     stream.Volume = _volume;
             }
         }

--- a/MonoGame.Framework/Utilities/OggStream.cs
+++ b/MonoGame.Framework/Utilities/OggStream.cs
@@ -52,6 +52,8 @@ namespace MonoGame.Utilities
         public Action FinishedAction { get; private set; }
         public int BufferCount { get; private set; }
 
+        private bool _isDisposed = false;
+
         public OggStream(string filename, Action finishedAction = null, int bufferCount = DefaultBufferCount) : this(File.OpenRead(filename), finishedAction, bufferCount) { }
         public OggStream(Stream stream, Action finishedAction = null, int bufferCount = DefaultBufferCount)
         {
@@ -158,6 +160,9 @@ namespace MonoGame.Utilities
 
         public void Stop()
         {
+            if (_isDisposed)
+                return;
+
             var state = AL.GetSourceState(alSourceId);
             if (state == ALSourceState.Playing || state == ALSourceState.Paused)
                 StopPlayback();
@@ -243,8 +248,10 @@ namespace MonoGame.Utilities
 
             if (ALHelper.Efx.IsInitialized)
                 ALHelper.Efx.DeleteFilter(alFilterId);
-
+            
             ALHelper.Check();
+
+            _isDisposed = true;
         }
 
         void StopPlayback()

--- a/MonoGame.Framework/Utilities/OggStream.cs
+++ b/MonoGame.Framework/Utilities/OggStream.cs
@@ -52,8 +52,6 @@ namespace MonoGame.Utilities
         public Action FinishedAction { get; private set; }
         public int BufferCount { get; private set; }
 
-        private bool _isDisposed = false;
-
         public OggStream(string filename, Action finishedAction = null, int bufferCount = DefaultBufferCount) : this(File.OpenRead(filename), finishedAction, bufferCount) { }
         public OggStream(Stream stream, Action finishedAction = null, int bufferCount = DefaultBufferCount)
         {
@@ -160,9 +158,6 @@ namespace MonoGame.Utilities
 
         public void Stop()
         {
-            if (_isDisposed)
-                return;
-
             var state = AL.GetSourceState(alSourceId);
             if (state == ALSourceState.Playing || state == ALSourceState.Paused)
                 StopPlayback();
@@ -250,8 +245,6 @@ namespace MonoGame.Utilities
                 ALHelper.Efx.DeleteFilter(alFilterId);
             
             ALHelper.Check();
-
-            _isDisposed = true;
         }
 
         void StopPlayback()

